client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/operations/IterationNextOperation.java

```diff
@@ -90,9 +90,8 @@ protected IterationNextOperation(Codec codec, int flags, byte[] cacheName, Atomi
                entries.add(new SimpleEntry<>(unmarshall(key, status), (E) value));
             }
          }
-         segmentKeyTracker.segmentsFinished(finishedSegments);
-
       }
+      segmentKeyTracker.segmentsFinished(finishedSegments);
```

Comment:
This is to allow the client to receive completed segments even if no entries were pulled out of the remote iteration